src/webhook/webhook-receiver.ts

```diff
@@ -0,0 +1,80 @@
+import { Request, Response } from "express";
+import { verify } from "@octokit/webhooks-methods";
+import { GitHubServerApp } from "../models/git-hub-server-app";
+import { getPayload } from "./get-payload";
+import { Webhooks } from "./webhooks";
+import { getLogger } from "../config/logger";
+import { WebhookContext } from "./types";
+
+const WEBHOOK_HEADERS = [
+	"x-github-event",
+	"x-hub-signature-256",
+	"x-github-delivery",
+];
+
+export const webhookReceiver = async (webhooks: Webhooks, request: Request, response: Response) => {
+	const missingHeaders = (WEBHOOK_HEADERS.filter((header) => !(header in request.headers))).join(", ");
+	if (missingHeaders) {
+		response.writeHead(400, {
+			"content-type": "application/json",
+		});
+		response.end(`Required headers missing: ${missingHeaders}`);
+		return;
+	}
+	// GitHub will abort the request if it does not receive a response within 10s
+	let didTimeout = false;
+	const timeout = setTimeout(() => {
+		didTimeout = true;
+		response.statusCode = 202;
+		response.end("still processing\n");
+	}, 9000).unref();
```

Comment:
Why are we setting and clearing timeouts everywhere?

Reply:
GitHub will abort the request if it does not receive a response within 10s. So, setting timeout to response within that timeframe if our handler took more time to response and clearing timeout if event handler responds to the request.

Reply:
github doesn't need to wait for our processing, it's just waiting for a confirmation that we received the event.  We can response right away with a '204' while we're still processing the payload in the background.

Reply:
I'm also kinda curious as to what would be taking 10s to process...

Reply:
It is to handle edge use cases where it takes more than 10 secs and I found this in @octokit/webhooks source code which is being used by Probot. If we're not wait for processing to complete before sending response to github, then I'll remove this code as you mentioned earlier.

Reply:
What could be a use-case when we handle a request more than 10 seconds? Shouldn't it be handled asynchronously if such is possible at all?

Reply:
Updated flow to send response back to incoming webhook request without waiting for processing to complete. Removed timeout statements as well.